src/core/style/expressions/buckets.js

```diff
@@ -4,10 +4,59 @@ import { implicitCast } from './utils';
 let bucketUID = 0;
 
 export default class Buckets extends Expression {
-    /*
-        If input is numeric => args is a breakpoint list
-        If input is categorical => args is a list of category names to map input
-    */
+
+    /**
+     * Given a property create "sub-groups" based on the given breakpoints.
+     * 
+     * 
+     * Imagine a trafic dataset with a speed property. We want to divide the roads in
+     * 3 buckets (slow, medium, fast) based on the speed using a different color each bucket.
+     * 
+     * 
+     * We´ll need:
+     *  - A {@link carto.style.expressions.ramp|ramp } to add a color for every bucket.
+     *  - A {@link carto.style.expressions.palettes|colorPalette } to define de color scheme.
+     * 
+     * 
+     * ```javascript
+     *  const s = carto.style.expressions;
+     *  const $speed = s.prop('speed');
+     *  const style = new carto.Style({
+     *  color: s.ramp(
+     *      s.buckets($speed, 30, 80, 120),
+     *      s.palettes.PRISM),
+     * });
+     * ```
+     * 
+     * Using the buckets `expression` we divide the dataset in 3 buckets according to the speed:
+     *  - From 0 to 29 
+     *  - From 30 to 79
+     *  - From 80 to 120
+     * 
+     * Values lower than 0 will be in the first bucket and values higher than 120 will be in the third one.
+     *
+     * This expression can be used for categorical properties, imagine the previous example with the data already
+     * procesed in a new categorical `procesedSpeed` column:
+     * 
+     * ```javascript
+     *  const s = carto.style.expressions;
+     *  const $procesedSpeed = s.prop('procesedSpeed');
+     *  const style = new carto.Style({
+     *  color: s.ramp(
+     *      s.buckets($procesedSpeed, 'slow', 'medium', 'high'),
+     *      s.palettes.PRISM),
+     * });
+     * ```
+     * 
+     * @param {carto.style.expressions.property} property - The property to be evaluated and interpolated
+     * @param {...carto.style.expressions.expression} breakpoints - Numeric expression containing the different breakpoints.
```

Comment:
Is this `carto.style.expressions.property` or `carto.style.expressions.prop`.
Why the 3 dots?

Reply:
both are valid, `s.prop` is an alias for `s.property`.

... dots means you can pass n arguments